components/sup/src/manager/service.rs

```diff
@@ -1217,7 +1252,6 @@ impl<'a> Serialize for ServiceProxy<'a> {
         strukt.serialize_field("desired_state", &s.desired_state)?;
         strukt.serialize_field("health_check", &s.health_check_result)?;
         strukt.serialize_field("hooks", &s.hooks)?;
-        strukt.serialize_field("initialized", &s.initialized)?;
```

Comment:
Is this needed? It looks as if we just check if the process is up and use that to determine if the process was initialized.

Reply:
Yes and no :frowning_face: 

The fact that this field is here at all is because waaaaay back in the beginning, the HTTP gateway basically made direct JSON serializations of internal datastructures available to users. I can't really think of a legitimate use for this field in such a situation, but it _is_ available to users.

For additional background, check out https://github.com/habitat-sh/habitat/pull/5689 as well as its spiritual predecessor https://github.com/habitat-sh/habitat/pull/4823. At some point I'd like to do something like https://github.com/habitat-sh/habitat/issues/4872, but for the HTTP gateway data.

For now, though, to preserve existing behavior, this should turn into something like

```rust
strukt.serialize_field("initialized", &s.state.initialized())?;
```